libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/useridentity/service/UserIdentityService.java

```
@@ -496,114 +487,25 @@ public Optional<List<PostalAddress>> generateAddressClaim(List<VerifiableCredent
-        if (passportVc.get().getCredential()
-                instanceof IdentityCheckCredential identityCheckCredential) {
-
-            var credentialSubject = getIdentityCheckSubjectOrThrowError(identityCheckCredential);
-
-            var passport = credentialSubject.getPassport();
-
-            if (isNullOrEmpty(passport)) {
-                return Optional.empty();
-            }
-
-            return Optional.of(passport);
-        } else {
-            LOGGER.warn(
-                    LogHelper.buildLogMessage(MUST_BE_IDENTITYCHECK_MESSAGE)
-                            .with(
-                                    LOG_CRI_ISSUER.getFieldName(),
-                                    passportVc.get().getCri().getId()));
-            return Optional.empty();
-        }
-    }
-
-    private Optional<List<DrivingPermitDetails>> generateDrivingPermitClaim(
-            List<VerifiableCredential> verifiableCredentials)
-            throws HttpResponseExceptionWithErrorBody {
-        var drivingPermitVc = findVc(DRIVING_PERMIT_CRI_TYPES, verifiableCredentials);
-
-        if (drivingPermitVc.isEmpty()) {
-            return Optional.empty();
-        }
-
-        if (drivingPermitVc.get().getCredential()
-                instanceof IdentityCheckCredential identityCheckCredential) {
-
-            var credentialSubject = getIdentityCheckSubjectOrThrowError(identityCheckCredential);
-
-            var drivingPermit = credentialSubject.getDrivingPermit();
+    public <T> Optional<List<T>> getFirstClaim(
+            List<VerifiableCredential> vcs, Function<IdentityCheckSubject, List<T>> getClaim) {
+        for (var vc : vcs) {
+            if (vc.getCredential() instanceof IdentityCheckCredential identityCheckCredential) {
+                var credentialSubject = identityCheckCredential.getCredentialSubject();
 
-            if (isNullOrEmpty(drivingPermit)) {
-                return Optional.empty();
+                var claim = getClaim.apply(credentialSubject);
```

Based on the [schema](https://vocab.account.gov.uk/v1/json-schemas/IdentityCheckCredential.json), it looks like the credential subject can be null. Does this mean that we can have no passport or DL claims? If so, should we log something?

---

We were previously handling this by throwing an error since we needed the credential subject to get the required details but not sure if that's needed since we don't do the same for the `identityClaim` cc @blakeyp @Joe-Edwards-GDS

---

It does need to be able to work with `null` `credentialSubject`'s, good spot thank you! We might want logging, though I thought throwing an error wasn't consistent with the earlier checks which instead return `Optional.empty()`

---

We wouldn't expect a DL/passport VC without credentialSubject, we should probably still check and throw if it's missing

---

Should they also produce claims? If so, that would be a more obvious check to use